packages/backend/aws/src/cognito/update-cognito-user-groups.ts

```diff
@@ -0,0 +1,107 @@
+import * as Joi from "joi";
+import * as R from "ramda";
+import { jsonApiError } from "@gravitywelluk/json-api-error";
+import type AWSModule from "aws-sdk";
+
+import { awsError } from "../utils";
+import { cognitoConfigure } from "./cognito-configure";
+
+export interface UpdateCognitoGroupsParams {
+  userPoolId: string;
+  cognitoId: string;
+  groups: string[];
+}
+
+/**
+ * Updates the given Cognito user's groups, equivalent to a PUT request
+ *
+ * @param updateUserParams - The parameters required to create a Cognito user
+ * @param awsCognitoConfigOverrides - Configuration option overrides
+ */
+export const updateCognitoUserGroup = async (
+  updateUserParams: UpdateCognitoGroupsParams,
+  awsCognitoConfigOverrides: AWSModule.CognitoIdentityServiceProvider.ClientConfiguration = {}
+): Promise<AWSModule.CognitoIdentityServiceProvider.GroupListType> => {
+  const cognito = cognitoConfigure(awsCognitoConfigOverrides);
+  const requestedGroups = updateUserParams.groups || [];
+
+  const { error } = Joi.object({
+    userPoolId: Joi.string().required(),
+    cognitoId: Joi.string().required(),
+    groups: Joi.array().items(Joi.string().required()).required()
+  }).validate(updateUserParams);
+
+  let allCognitoGroups: string[] = [];
+
+  // Error if there any Joi validation errors
+  if (error) {
+    throw jsonApiError(error);
+  }
+
+  // Get the current Cognito groups with the given user pool
+  try {
+    const cognitoGroupList = await cognito.listGroups({ UserPoolId: updateUserParams.userPoolId }).promise();
+
+    // If groups are returned in the response, collate the group names and set
+    // allCognitoGroups
+    allCognitoGroups = cognitoGroupList.Groups ? cognitoGroupList.Groups.map(group => group.GroupName).filter(groupName => typeof groupName === "string") as string[] : [];
+  } catch (error) {
+    throw awsError(error, {
+      environment: process.env.ENVIRONMENT,
+      functionName: "updateCognitoUserGroups"
+    });
+  }
```

Comment:
We're using this logic in both create and update—could we move this into a utility function?
```
const getAvailableCognitoGroups = async (userPoolId: string) => string[];
```